proxy-node-shared/src/main/java/uk/gov/ida/notification/validations/Base64SamlValidator.java

```diff
@@ -0,0 +1,36 @@
+package uk.gov.ida.notification.validations;
+
+import org.apache.commons.lang3.StringUtils;
+import org.glassfish.jersey.internal.util.Base64;
+import org.opensaml.saml.saml2.core.impl.ResponseImpl;
+import uk.gov.ida.notification.saml.SamlParser;
+
+import javax.validation.ConstraintValidator;
+import javax.validation.ConstraintValidatorContext;
+
+public class Base64SamlValidator implements ConstraintValidator<ValidBase64Saml, String> {
+
+    public static final int MinLength = 64;
+    public static final int MaxLength = 64000;
+
+    @Override
+    public void initialize(ValidBase64Saml constraint) { /* intentionally blank */ }
+
+    @Override
+    public boolean isValid(String value, ConstraintValidatorContext context) {
+        if (StringUtils.isBlank(value)) { return true; } // @NotNull should detect nulls
+
+        try {
+            String decoded = Base64.decodeAsString(value);
+            if (StringUtils.isBlank(decoded)) {
+                return false;
+            }
+
+            Object saml = new SamlParser().parseSamlString(decoded);
+            return saml instanceof ResponseImpl;
+
+        } catch (Exception e) {
```

Comment:
Can we be less generic with the exception?

Reply:
The intention here was to catch _anything_ that goes wrong when trying to parse the String. A generic Exception seemed to capture that intention more clearly. Happy to change it if you think it's better to use the specific subclasses.

Reply:
None of the code here actually throws an exception. As we're inside a validator, my feeling is that we should capture any runtime exceptions, and return false for anything unexpected that means it can't resolve the Base64 String into SAML. I'm going to leave it in, and comment my decision.